tiledb/storage_format/uri/generate_uri.cc

```diff
@@ -38,11 +38,9 @@ namespace tiledb::storage_format {
 
 std::string generate_uri(
```

Comment:
This is purely a comment that you can feel free to ignore, but given that we're already planning on rerouting all the places that generate URIs through this function it might be beneficial to do that as part of this PR since we're already modifying the places that would be calling into it. From my spot checking you'll need to add a version that doesn't append the version on but that's pretty trivial.

I didn't look hard enough, but the other thing is a version that takes a `std::pair<uint64_t, uint64_t>` and proxies through to the existing one might be helpful but might be not.

Reply:
Thanks, I've been asked before to unify these calls, will do as part of this PR.